### DISCUSSION TITLE
Models without morphs throws ClassMorphViolationException

### DISCUSSION THREAD

**Discussion by tarkis** [livewire/livewire#6862] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Have an enforceMorphMap() in your AppServiceProvider
```
Relation::enforceMorphMap([
    'team' => Team::class,
]);
```
2. Create a Livewire component with other Model that does not have morph.
```
class User extends Component
{
    public User $user;

    public function mount(): void
    {
        $this->user = User::find(1);
    }
}
```
3. Load this component into your balde and you will get `No morph map defined for model [App\Models\User].`

What was the reason to use `$target->getMorphClass();` at https://github.com/livewire/livewire/blob/main/src/Features/SupportModels/ModelSynth.php#L25? 

Looking at Laravel code [this function](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php#L766) should be called to get a class name for polymorphic relations. Right now I'm being forced to write morphs for all my models that I will use with Livewire due to me having `enforceMorphMap()` in my AppServiceProvider and [this if throwing an exception](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php#L779)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by AxelGuignard** [comment]
I have the same issue, and came to the exact same conclusion. Why do we call getMorphClass() in ModelSynth and (in my case) EloquentCollectionSynth?
A workaround is to use Relation::morphMap instead of Relation::enforceMorphMap in AppServiceProvider, but I don't like solutions that imply to reject a useful functionality.
It's weird that this issue don't have more visibility given the original post is from 2 years ago and it impacts a core functionality of Laravel...

**Reply by Akhmami** [reply]
Same here, is this bug?

**Comment by calebporzio** [comment, marked as answer]
Fixed, will be in the next release. #9748
